apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/manager/room/LocalParticipantWrapper.kt

```diff
@@ -0,0 +1,182 @@
+        if (!isAudioMuted) setupLocalAudioTrack()
+        if (!isVideoMuted) setupLocalVideoTrack()
+    }
+
+    override fun onPause(owner: LifecycleOwner) {
+        super.onPause(owner)
+        removeCameraTrack()
+    }
+
+    fun toggleLocalAudio() {
+        if (!isAudioMuted) {
+            isAudioMuted = true
+            removeAudioTrack()
+        } else {
+            isAudioMuted = false
+            setupLocalAudioTrack()
+        }
+    }
+
+    fun toggleLocalVideo() {
+        if (!isVideoMuted) {
+            isVideoMuted = true
+            removeCameraTrack()
+        } else {
+            isVideoMuted = false
+            setupLocalVideoTrack()
+        }
+    }
+
+    private fun setupLocalAudioTrack() {
+        if (localAudioTrack == null && !isAudioMuted && context != null) {
+            localAudioTrack = createLocalAudioTrack(context, true, MICROPHONE_TRACK_NAME)
+            localAudioTrack?.let { publishAudioTrack(it) }
+                ?: Timber.e(RuntimeException(), "Failed to create local audio track")
+        }
+    }
+
+    private fun setupLocalVideoTrack() {
+        context?.let {
+            cameraCapturer = CameraCapturerCompat.newInstance(it)
+
+            videoTrack = cameraCapturer?.let { capturer ->
+                LocalVideoTrack.create(it, true, capturer, null, CAMERA_TRACK_NAME)
+            }?.apply {
+                localVideoTrackNames[name] = context.getString(R.string.camera_video_track)
+                publishCameraTrack(this)
+            }
+        }
+    }
+
+    private fun publishCameraTrack(localVideoTrack: LocalVideoTrack?) {
+        if (!isVideoMuted) {
+            localVideoTrack?.let {
+                participant?.publishTrack(
+                    it,
+                    LocalTrackPublicationOptions(TrackPriority.LOW)
+                )
+            }
+        }
+    }
+
+    private fun publishAudioTrack(localAudioTrack: LocalAudioTrack?) {
+        if (!isAudioMuted) {
+            localAudioTrack?.let { participant?.publishTrack(it) }
+        }
+    }
+
+    private fun unPublishTrack(localVideoTrack: LocalVideoTrack?) =
+        localVideoTrack?.let { participant?.unpublishTrack(it) }
+
+    private fun unPublishTrack(localAudioTrack: LocalAudioTrack?) =
+        localAudioTrack?.let { participant?.unpublishTrack(it) }
+
+    private fun removeCameraTrack() {
+        videoTrack?.let { cameraVideoTrack ->
+            unPublishTrack(cameraVideoTrack)
+            localVideoTrackNames.remove(cameraVideoTrack.name)
+            cameraVideoTrack.release()
+            this.videoTrack = null
+        }
+    }
+
+    private fun removeAudioTrack() {
+        localAudioTrack?.let { localAudioTrack ->
+            unPublishTrack(localAudioTrack)
+            localAudioTrack.release()
+            this.localAudioTrack = null
+        }
+    }
+
+    override fun onNetworkQualityLevelChanged(localParticipant: LocalParticipant, networkQualityLevel: NetworkQualityLevel) {
```

Comment:
The iOS app does not use this and so I would recommend removing it for Android.

Reply:
Ohhh ok, I implemented this very easy ways, because I see in the `AndroidVideoRoomApp` it was implemented, so I thought would it be nice to have this, just in case for future.

Reply:
I would recommend only including code that we really need in order to minimize maintenance and make the code as simple as possible to follow for customers.

Reply:
Done!